### DISCUSSION TITLE
how can i check the identities i created on infura using the nodejs veramo api

### DISCUSSION THREAD

**Discussion by anotherrohit** [decentralized-identity/veramo#365] [General]
i am looking to create identifiers using the nodejs example from veramo docs. I was able to create and verify the ids in the app, but how can i confirm they were created on eth? Infura does not show these ids.

**Comment by mirceanis** [comment]
I'm assuming the identifiers you created are of the `did:ethr` type.
The `did:ethr` protocol does not involve any transaction when creating new ids. This is by design, to reduce the burden on the ethereum network, to make onboarding free, and  and to allow the use-case of disposable identifiers.

These identifiers don't appear anywhere on-chain until you try to modify the corresponding DID document using methods like `didManagerAddKey()`, `didManagerAddService()`, etc...
You will notice that these methods require `gas` to complete. This is because any transaction on a public ethereum network needs to be funded. The `did:ethr` protocol supports meta-transactions, in which case funding can come from a 3rd party, but this is not implemented at the moment in veramo.

You can find out more by going through the [ethr DID method spec](https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md).

Note that `did:ethr` is only one of the possible DID providers usable in veramo.
There is also a `did:web` provider and the next feature version will have a `did:key` implementation.
There are plans to add `did:ion` and `did:elem` providers as well and, of course, DID provider plugins could be implemented by other parties, which we would gladly showcase.

edit:
corrected `didManager*` method names

**Reply by anotherrohit** [reply]
What a fantastic answer! can you show me some examples of addKey and addService. Also how can I add gas to my transactions using veramo?

**Reply by anotherrohit** [reply]
for resolving an exisitng registry address { rpcUrl: 'https://rinkeby.infura.io/ethr-did', registry: registry.address } where can i get this address for myself?

**Reply by mirceanis** [reply]
The registry addresses for the most popular public networks are listed in the table from the [ethr-DID method spec](https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md#abstract)
You'll notice that most public networks use the [0xdca7ef03e98e0dc2b855be647c39abe984fcf21b](https://etherscan.io/address/0xdca7ef03e98e0dc2b855be647c39abe984fcf21b) address.
Unless you're doing research about this DID method, it's not recommended to use a different registry address because then the DIDs managed there will not be resolvable without knowing your contract.

That being said, your question made me realize that there is an oversight in the ethr-did-resolver documentation about the infura URL, thank you for spotting it!
If you want to use the ethr-did-resolver directly, with an infura RPC, you can get a free project ID from https://infura.io
Then, perhaps use this [example](https://github.com/uport-project/veramo/blob/92a86d6f8cf652e731ca662085efe78aeab198eb/__tests__/localAgent.test.ts#L125) of instantiating the resolver.

Also, make sure to go through the [veramo docs](https://veramo.io/docs/fundamentals/introduction) to hopefully get a better understanding of what this project is about.
We realize that the concepts juggled here are not straightforward and most of the time not very well-documented so we appreciate any input regarding confusing docs or broken sample code.

**Reply by anotherrohit** [reply]
@mirceanis I am trying to create a registration page with updates of personal attributes while creating DIDs and JWT and also a resolver that can validate the did with the JWT created above. Can you suggest what blocks of code I can use from the examples?   I scoured through the documentation but am still lost on how to implement this requirement.

**Reply by mirceanis** [reply]
I'm not sure I understand, so let me try to rephrase your requirements.
Do you want to have a way for people to express personal attributes as verifiable credentials?
In that case, I suppose each person would be the controller of a DID and then every attribute could be expressed as a JWT verifiable credential signed by that DID.

Is this what you are looking to create?

**Reply by anotherrohit** [reply]
@mirceanis that is exactly correct! I want people to express personal attributes as verifiable credentials. How best to implement this use case?

**Reply by mirceanis** [reply]
You are right, it seems that the guides on the docs site don't mention actually creating some credentials.
There is an issue about that missing guide here: https://github.com/uport-project/veramo-website/issues/44

Until that is fixed, you can look through the integration tests to get a rough idea.
* [configure an agent using all the plugins](https://github.com/uport-project/veramo/blob/57fa00393e5268e57c2b409952f3181f1f6a7a13/__tests__/localAgent.test.ts#L79
), including the one for [credential creation](https://github.com/uport-project/veramo/blob/57fa00393e5268e57c2b409952f3181f1f6a7a13/__tests__/localAgent.test.ts#L149)
* use the agent to [create a DID](https://github.com/uport-project/veramo/blob/57fa00393e5268e57c2b409952f3181f1f6a7a13/__tests__/shared/verifiableData.ts#L23)
* use the agent to [create a credential signed by that DID](https://github.com/uport-project/veramo/blob/57fa00393e5268e57c2b409952f3181f1f6a7a13/__tests__/shared/verifiableData.ts#L57)

I hope this helps

**Reply by jasheal** [reply]
Also relates to this PR https://github.com/uport-project/veramo-website/pull/48
